sds_data_manager/stacks/api_gateway_stack.py

```diff
@@ -0,0 +1,89 @@
+# Installed
+from constructs import Construct
+from aws_cdk import (
+    Stack,
+    Environment,
+    aws_apigateway as apigw,
+    aws_route53 as route53,
+    aws_certificatemanager as acm,
+    aws_route53_targets as targets
+)
+
+
+class ApiGateway(Stack):
+    """Sets up api gateway, creates subdomains, and creates methods that
+    are linked to the lambda function"""
+
+    def __init__(self, scope: Construct,
+                 construct_id: str,
+                 sds_id: str,
+                 lambda_functions: dict,
+                 env: Environment,
+                 hosted_zone: route53.IHostedZone = None,
+                 certificate: acm.ICertificate = None,
+                 use_custom_domain: bool = False,
+                 **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        scope : Construct
+        construct_id : str
+        sds_id : str
+            Name suffix for stack
+        lambda_functions : dict
+            Lambda functions
+        env : Environment
+        hosted_zone : route53.IHostedZone
+            Hosted zone used for DNS routing.
+        certificate : acm.ICertificate
+            Used for validating the secure connections to API Gateway.
+        use_custom_domain : bool
+            Determines if a custom domain should be used.
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # Define subdomains
+        subdomains = lambda_functions.keys()
+
+        for subdomain in subdomains:
+            # Set up the API Gateway
+            api = apigw.RestApi(self, f'{subdomain}-RestApi-{sds_id}',
+                                rest_api_name=f'My {subdomain.capitalize()} Service',
+                                description=f'This service serves as my {subdomain.capitalize()} API Gateway.',
+                                deploy_options=apigw.StageOptions(stage_name='dev'),
+                                endpoint_types=[apigw.EndpointType.REGIONAL]
+                                )
+
+            # If using custom domain
+            if use_custom_domain:
+                # Define a custom domain
+                custom_domain = apigw.DomainName(self,
+                                                 f'{subdomain}-DomainName-{sds_id}',
+                                                 domain_name=f'{subdomain}.imap-mission.com',
+                                                 certificate=certificate,
+                                                 endpoint_type=apigw.EndpointType.REGIONAL
+                                                 )
+
+                # Route domain to api gateway
```

Comment:
Is this required? We don't use it on SWxTREC, so I'm not sure exactly what it is doing here. I couldn't quite grasp the AWS documentation either, so I'm mostly just curious what is going on.

Reply:
Here we link the custom domain (custom_domain) to the API (api). If we use the default AWS API Gateway URL, we don't need to use it. Does SWXTREC have a custom domain?